File: backend/Projekatsiit2023/src/main/java/rs/ac/uns/ftn/asd/Projekatsiit2023/services/RegisterService.java
package rs.ac.uns.ftn.asd.Projekatsiit2023.services;

import rs.ac.uns.ftn.asd.Projekatsiit2023.dto.request.RegisterRequest;
import rs.ac.uns.ftn.asd.Projekatsiit2023.dto.response.RegisterResponse;

public interface RegisterService {
    RegisterResponse register(RegisterRequest request);
    String generateActivationLink(Long userId);
}

File: web/taxi-app-web/src/app/services/register-service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { RegisterResponse } from '../models/register-response';
import { RegisterRequest } from '../models/register-request';

@Injectable({
  providedIn: 'root'
})
export class RegisterService {
  private readonly API_URL = 'http://localhost:8080/api/auth/register';

  constructor(private http: HttpClient) {}

  register(userData: RegisterRequest): Observable<RegisterResponse> {
    return this.http.post<RegisterResponse>(this.API_URL, userData);
  }
}